pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/SingleParamMathTransformFunction.java

```diff
@@ -0,0 +1,248 @@
+          String[] stringValues = _firstTransformFunction.transformToStringValuesSV(projectionBlock);
+          for (int i = 0; i < length; i++) {
+            _result[i] = transformString(stringValues[i]);
+          }
+          break;
+        default:
+          throw new UnsupportedOperationException();
+      }
+    }
+
+    return _result;
+  }
+
+  protected double transformInt(int value) {
+    return transformDouble(value);
+  }
+
+  protected double transformLong(long value) {
+    return transformDouble(value);
+  }
+
+  protected double transformFloat(float value) {
+    return transformDouble(value);
+  }
+
+  protected double transformDouble(double value) {
+    return transformDouble(value);
+  }
+
+  protected double transformString(String value) {
+    double doubleValue = Double.parseDouble(value);
+    return transformDouble(doubleValue);
+  }
+
+  public static class AbsTransformFunction extends SingleParamMathTransformFunction {
+    public static final String FUNCTION_NAME = "abs";
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    protected double transformInt(int value) {
+      return Math.abs(value);
+    }
+
+    @Override
+    protected double transformLong(long value) {
+      return Math.abs(value);
+    }
+
+    @Override
+    protected double transformFloat(float value) {
+      return Math.abs(value);
+    }
+
+    @Override
+    protected double transformDouble(double value) {
+      return Math.abs(value);
+    }
+  }
+
+  public static class CeilTransformFunction extends SingleParamMathTransformFunction {
+    public static final String FUNCTION_NAME = "ceil";
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    protected double transformDouble(double value) {
+      return Math.ceil(value);
+    }
+  }
+
+  public static class ExpTransformFunction extends SingleParamMathTransformFunction {
+    public static final String FUNCTION_NAME = "exp";
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    protected double transformDouble(double value) {
+      return Math.exp(value);
+    }
+  }
+
+  public static class FloorTransformFunction extends SingleParamMathTransformFunction {
+    public static final String FUNCTION_NAME = "floor";
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    protected double transformDouble(double value) {
+      return Math.floor(value);
+    }
+  }
+
+  public static class LnTransformFunction extends SingleParamMathTransformFunction {
+    public static final String FUNCTION_NAME = "ln";
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    protected double transformDouble(double value) {
+      return Math.log(value);
+    }
+  }
+
+  public static class Log10TransformFunction extends SingleParamMathTransformFunction {
```

Comment:
Instead of Log10 and ln, can we have log(number, base)?

Reply:
@mayankshriv , yes `log(number, base)` is more generic and better to have. This pr only related with kinds of single parameter function which are have common base class. Will try to add this in the following. Thank you~

Reply:
@xueyumusic In that case, could you please take out the ln and log10 from this PR? Otherwise we will end up with three different log functions.

Reply:
Done, @mayankshriv ,updated, thank you.